exporters/dsexporter/dsexporter_test.go

```diff
@@ -2,45 +2,136 @@ package main
 
 import (
 	"testing"
+	"reflect"
+	"net/http"
+	"net/http/httptest"
 
-	"github.com/prometheus/client_golang/prometheus"
 	"github.com/stretchr/testify/assert"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
 )
 
-func TestCustomCollector(t *testing.T) {
-	exporter := NewCustomCollector()
-	prometheus.MustRegister(exporter)
+func MockInclusterConfig() (*rest.Config) {
+    return nil
+}
+
+func MockNewForConfig(*rest.Config) (*kubernetes.Clientset) {
+	return nil
+}
+
+func TestNewKubeClient(t *testing.T) {
+	actualInclusterConfig := getInClusterConfig
+    getInClusterConfig = MockInclusterConfig
+    actualNewForConfig := getNewForConfig
+    getNewForConfig = MockNewForConfig
+
+    defer func() {
+        getInClusterConfig = actualInclusterConfig
+        getNewForConfig = actualNewForConfig
+    }()
 
-	// Simulate collecting metrics and check the exported metric value.
-	metrics := prometheus.DefaultGatherer
-	metricFamilies, err := metrics.Gather()
-	assert.NoError(t, err)
+    client := NewKubeClient()
```

Comment:
Once we drop all the error handling, I don't think we need a function for this, and I don't think it requires any tests. Can be just a one-liner.

```suggestion
    client := kubernetes.NewForConfigOrDie(config.GetConfigOrDie())
```
Also, the config function actually supports both local and in-cluster configs, so the name/comments might be a bit misleading.

Reply:
Updated.